fix(storybook): broken components (#2298)

andre-code opened the pull request:
This pull request handles broken stories and updates the affected file extensions from .stories-broken to the standard .stories extension.

### Testing:
- Tested each story to ensure it is functioning correctly and rendering as expected.
- Verified that the components in the stories are displaying the intended behavior.
- Confirmed that the updated files are correctly registered in Storybook and displayed in the Storybook UI.

Closes #2298 

/deploy renku=renku-ui/3.11-acceptance-tests #persist

lorenzo-cavazzi reviewed (changes requested):
This looks promising!

I would add a section in the Readme file to explain how to use Storybook:
- How do I start it?
- How do I develop it? Perhaps point to a couple of existing story files as examples.

Can we improve the existing stories? Here are a few examples:
- Having a meaningful list of defaults:
    - E.G. in the `AuthorFilter` it would be nice to have a dropdown list of values that are meaningful for the story. And perhaps also a working function so that we can interact -- not that we always need one, but it could be used as an example of how to make a good story
    ![image](https://github.com/SwissDataScienceCenter/renku-ui/assets/43481553/34ee899d-23c0-437a-8dbc-a39c2f7042b6)
- Provide more descriptions:
    - Buttons might give more information, i.e. what is the meaning of each button? When to use "Secondary Green" and when to use the others? Perhaps we could add descriptions.
    Also, when interacting, some of the content isn't visible (see screenshot -- the second element is open but I can't see the dropdown content). Is there a way to prevent this?
    ![image](https://github.com/SwissDataScienceCenter/renku-ui/assets/43481553/21cd9775-b03a-4c37-9f12-4225df8c421f)

It would be nice to categorize the components. Is there any convenient way to do that that you have seen in StoryBook examples? Otherwise, we would end up with a very long list of components and it might be hard to  find what we are looking for.

andre-code commented:
thanks for the review @lorenzo-cavazzi 😊

** Documentation: Definitely we should include the documentation with a good example, I included one in the README file.

** Existing stories: Yes, actually we should modify most of the Stories to use the new component 3 format, I created the [Issue](https://github.com/SwissDataScienceCenter/renku-ui/issues/2756) to also include that case since it would touch all the Stories and would be the best place to evaluate story improvements.

** Story Meaning Information: this is a great addition that we should work with the product-UI team as they define the guideline, I discussed this point with @jdupre81 so that once we have it published the Storybook they can review and request to add a more informative description if needed, in the meantime I have added descriptions for Button's story only.

** Dropdown button not well displayed: Fixed!

** Categorize Stories:  it is possible by modifying the title of the story, I have added a proposal of categories.

lorenzo-cavazzi reviewed (changes requested):
This looks amazing! Thank you for the updates. It really has a lot of potential :star_struck: 
I have a few additional change requests.

I pushed a change in the readme file to use just markdown instead of HTML elements. That prevents inconsistent indentation (see next screenshot; left was the previous version). I also adjusted the text a bit. Please feel free to add further changes if the text is not on point anymore.
![image](https://github.com/SwissDataScienceCenter/renku-ui/assets/43481553/24a54288-bc1b-42b9-b6be-d6f03b4a743e)

The categories work very well to group content together! This is precisely what I wished for :rocket:  It seems much easier to process information. 
Just a minor detail. In the Buttons menu, I noticed that variations like "ButtonWithMenu" and "RoundButtonGroup" are inside the "Buttons". Since there are a few examples of buttons, the last elements are harder to notice. Can we have them all at the same level?
I.E. switching from this structure:

- Buttons
  - ButtonsWithMenu
  - RoundButtonGroup

To something like this:

- Buttons
  - Normal Buttons
  - ButtonsWithMenu
  - RoundButtonGroup

EDIT: I see the Search section already has the desired structure. Could you make the buttons work similarly, with a common ancestor folder and all the specific components as children?

![image](https://github.com/SwissDataScienceCenter/renku-ui/assets/43481553/53ee4d72-876b-439a-a83b-3b389a1674c4)

I also added a few inline comments just for small details and rewording. Sorry for being so nitpicking with the Buttons section; I feel having all the details figured out there would be good since it will be our main example.

andre-code commented on client/src/components/buttons/Buttons.stories.tsx:
```diff
@@ -19,116 +19,195 @@ import * as React from "react";
+  - **Entities:** Projects, Datasets y Workflows
+  - **States:** Error, Information, Alert
+ 
+<details>
+<summary>Best Practices</summary>
+<ul>
+<li>For modals and single-page forms right-align buttons with the container.</li>
+<li>Always place the primary button on the right, the secondary button just to the left of it.</li>
+</ul>
+</details>
```
I'm more inclined towards using dropdowns instead of markdown for these texts. Given that the content might get quite extensive, having a quick and easy visual overview of the component would be more user-friendly. If someone is interested, they can simply open the accordion to delve into the best practice suggestions.

lorenzo-cavazzi replied:
Mixing Markdown and HTML frequently doesn't play out very well; this is a good example.
I assume you never meant to have the `<details>` section as a child of the second list item where you mention the States. Yet, this is what I get in the rendering

![image](https://github.com/SwissDataScienceCenter/renku-ui/assets/43481553/6a3f73a4-525c-434b-baea-5ea778eee0cc)

These things are tedious and render better or worse depending on the browser.
I suggest sticking to Markdown because it seems easier and quicker to write. If you _really_ want to have the `<details>` component, then let's convert all the description content to HTML to prevent these issues.

andre-code commented on client/src/components/buttons/Buttons.stories.tsx:
```diff
@@ -19,116 +19,195 @@ import * as React from "react";
+  docs: {
+    description: {
+      story:
+        'When employing a button for a non-entity-associated action that is not obligatory, apply the "btn-info" className.',
```
I appreciate the suggestion. However, the phrase "when suggesting a minor project version update" can be a bit confusing, as it shouldn't be tied directly to an entity like a project. I've made a slight modification to your suggestion:

Use "btn-info" in the context of an Info Alert, , or in situations where it's not strictly tied to a specific entity."

lorenzo-cavazzi replied:
Your version works well :+1:

andre-code commented:
Hey @lorenzo-cavazzi, appreciate your review! I've incorporated your suggestions into the text, and the README is now more descriptive. As for the dropdown section usage, I've shared my thoughts in a comment. Let's catch up during our upcoming daily meeting to discuss that part further. Thanks again for your input!

lorenzo-cavazzi commented on client/src/components/buttons/Buttons.stories.tsx:
```diff
@@ -19,116 +19,198 @@
+  docs: {
+    description: {
+      story:
+        "Use `btn-danger` in the context of a Danger Alert, or when suggesting a high-risk actions action that can lead to critical or irreversible outcomes. (E.G. `Delete Project` or updating a project when it is on a very outdated version not supported anymore by the UI).",
```
An additional `actions` slipped in (or `action`, one of the two)
```suggestion
        "Use `btn-danger` in the context of a Danger Alert, or when suggesting a high-risk action that can lead to critical or irreversible outcomes. (E.G. `Delete Project` or updating a project when it is on a very outdated version not supported anymore by the UI).",
```

lorenzo-cavazzi reviewed (approved):
Lgtm :rocket:
